src/components/common/FileSelect/FileItem.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import FileModel from 'models/FileSelect/FileModel'
+import ArbitraryNoticeModel from 'models/notices/ArbitraryNoticeModel'
+import { Translate } from 'react-redux-i18n'
+import { CircularProgress } from 'material-ui'
+import { download } from 'redux/ui/ipfs'
+import { ActionDelete, FileFileDownload } from 'material-ui/svg-icons'
+import { notify } from 'redux/notifier/actions'
+import FileIcon from './FileIcon'
+import globalStyles from 'styles'
+import './FileItem.scss'
+
+class FileItem extends Component {
+  static propTypes = {
+    file: PropTypes.instanceOf(FileModel),
+    onRemove: PropTypes.func.isRequired,
+    handleDownload: PropTypes.func
+  }
+
+  renderErrors () {
+    const errors: Array = this.props.file.error()
+    return errors.length > 0
+      ? (
+        <div styleName='errors'>
+          {errors.map((item, i) => {
+            const value = typeof item === 'string' ? {value: item} : item
+            return <div key={i} styleName='error'><Translate {...value} /></div>
+          })}
+        </div>
+      )
+      : null
+  }
+
+  renderButtons (file: FileModel) {
+    if (file.uploading()) {
+      return <CircularProgress size={16} thickness={1.5} />
+    }
+    return (
+      <div styleName='actionButtons'>
+        {file.hasErrors()
+          ? null
+          : (
+            <FileFileDownload
+              styleName='buttonItem'
+              onTouchTap={() => this.props.handleDownload(file.hash(), file.name())} />
+          )
+        }
+        {file.hasErrors() || file.uploaded()
```

Comment:
flag uploaded must be true if we fetch file from ipfs

Reply:
Done